app/models/ability.rb

```diff
@@ -0,0 +1,14 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new # Guest user
+
+    if user.role == 'admin'
+      can :manage, :all
+    else
+      can :read, :all
+      can :delete, [Post, Comment], author_id: user.id
```

Comment:
- [ ] Excellent work implementing authorizations for a `user`. Kindly add the conditions to `manage` a `Comment` that belongs to a `user`.